function_computer_grid.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed May 13 13:16:43 2020
+
+@author: carolinstreitberger
+"""
+import numpy as np
+import random
+
+def generate_random_position(size_ship):
+    """
+    This function will generate the position of a ship of a given size.
+
+    Parameters
+    ----------
+    size_ship : TYPE #?
+        Size of the ship whose position will be determined.
+
+    Returns
+    -------
+    orientation : TYPE
+        Vertical or horizontal.
+    y_start_position : TYPE
+        Starting point of the ship on the y-axis.
+    x_start_position : TYPE
+        Starting point of the ship on the x-axis.
+    """
+    orientation = random.randrange(0, 2, 1) # 0 is vertical and 1 is horizontal orientation
+    if (orientation == 0):
+        y_start_position = random.randrange(1, 11 - size_ship, 1)
+        # 11 - size_ship so that the ship is never bigger than 10
+        x_start_position = random.randrange(1, 11, 1)
+        # randrange is exclusive, ie 10 is never selected
+        # randrange is from 1 to 11 (so any number between 1 and 10) because 0 and 11 will be deleted later
+    elif (orientation == 1):
+        y_start_position = random.randrange(1, 11, 1)
+        x_start_position = random.randrange(1, 11 - size_ship, 1)
+    
```

Comment:
A smart way of making sure that the ships are not outside the grid!

Reply:
I always like in code to see where numbers come from. However, I am not sure whether this is considered good programming practice. For example, now I see random.randrange(1, 11, 1). I assume the 11 stands for how large the grid can be. It would be nice if you could use a variable instead of 11.

Reply:
I see what you mean but then this is what the code would look like: random.randrange(1, initial_grid.shape[0] - 1 - size_ship, 1). It has to be initial_grid.shape[0] - 1 because I made a grid that has two rows and two columns too many but I need them for the check_position function. (The initial_grid.shape = 12x12 and the final grid.shape = 10x10).